### ISSUE TITLE
Validate latests macOS codenames

### ISSUE THREAD

**Issue by jnasselle** [wazuh/wazuh-qa#3000]
| Related issue |
|---|
| wazuh/wazuh#13836|

# Description
This issue aims to validate wazuh/wazuh#13836 implementation by checking retrieved OS codenames from latest macOS versions (BigSur, Monterey, Ventura).

# Test cases
## Steps
- Install macOS agent on affected OS (BigSur, Monterey, Ventura) and connect it to Wazuh Manager
## Expected
- macOS codename should be correctly shown on: API endpoints ([syscollector](https://documentation.wazuh.com/current/user-manual/api/reference.html#operation/api.controllers.syscollector_controller.get_os_info), [agents](https://documentation.wazuh.com/current/user-manual/api/reference.html#operation/api.controllers.agent_controller.get_agent_summary_os)), CLI tools ([agent_control](https://documentation.wazuh.com/current/user-manual/reference/tools/agent-control.html)) and Wazuh Dashboard.
## Current
- macOS codename does not exist on Inventory Data or is shown as ''Unknown" on CLI tools

## Configuration details
PR: https://github.com/wazuh/wazuh/pull/13837
| Wazuh Commit | Agent Package | Manager Package|
|--|--|--|
| https://github.com/wazuh/wazuh/pull/13837/commits/c579f85577d4cf9dec9969f7d751040f9824f12d | https://packages-dev.wazuh.com/warehouse/pullrequests/4.3/rpm/var/wazuh-manager-4.3.4-0.commitc579f85.x86_64.rpm | https://packages-dev.wazuh.com/warehouse/pullrequests/4.3/macos/wazuh-agent-4.3.4-0.commitc579f85.pkg |

**Comment by CamiRomero** [comment]
Moved to On Hold until Issue [846](https://github.com/wazuh/wazuh-automation/issues/846) is ready

**Comment by CamiRomero** [comment]
## Test cases Monterey

<details>
<summary>API endpoints syscollector :green_circle: </summary>

1. Get token by API:
    ```
    curl -u wazuh:wazuh -k -X GET "https://<HOST_IP>:55000/security/user/authenticate"
    ```
2. Get agent OS:
    ```
    curl --location --request GET 'https://<HOST_IP>:55000/syscollector/001/os' --header TOKEN
    ```

    Output: 
        
    ```
    {
        "data": {
            "affected_items": [
                {
                    "os": {
                        "build": "21A559",
                        "codename": "Monterey",
                        "major": "12",
                        "minor": "0",
                        "name": "macOS",
                        "platform": "darwin",
                        "version": "12.0.1"
                    },
                    "scan": {
                        "id": 0,
                        "time": "2022-06-23T19:14:45+00:00"
                    },
                    "release": "21.1.0",
                    "architecture": "x86_64",
                    "hostname": "macos-1201",
                    "sysname": "Darwin",
                    "version": "Darwin Kernel Version 21.1.0: Wed Oct 13 17:33:23 PDT 2021; root:xnu-8019.41.5~1/RELEASE_X86_64",
                    "agent_id": "002"
                }
            ],
            "total_affected_items": 1,
            "total_failed_items": 0,
            "failed_items": []
        },
        "message": "All specified syscollector information was returned",
        "error": 0
    }
    ```
</details>

<details>
<summary>API endpoints agents :green_circle: </summary>

1. Get API token:
    ```
    curl -u wazuh:wazuh -k -X GET "https://<HOST_IP>:55000/security/user/authenticate"
    ```
2. Get agent OS:

    ```
    {
                    "os": {
                        "arch": "x86_64",
                        "codename": "Monterey",
                        "major": "12",
                        "minor": "0",
                        "name": "macOS",
                        "platform": "darwin",
                        "uname": "Darwin |macos-1201 |21.1.0 |Darwin Kernel Version 21.1.0: Wed Oct 13 17:33:23 PDT 2021; root:xnu-8019.41.5~1/RELEASE_X86_64 |x86_64",
                        "version": "12.0.1"
                    },
                    "group": [
                        "default"
                    ],
                    "manager": "ip-172-31-6-172.ec2.internal",
                    "ip": "FE80:0000:0000:0000:1C83:439E:A704:7490",
                    "id": "002",
                    "name": "macos-1201",
                    "configSum": "ab73af41699f13fdd81903b5f23d8d00",
                    "status": "active",
                    "version": "Wazuh v4.4.0",
                    "lastKeepAlive": "2022-06-23T19:23:22+00:00",
                    "registerIP": "any",
                    "mergedSum": "4a8724b20dee0124ff9656783c490c4e",
                    "node_name": "node01",
                    "group_config_status": "synced",
                    "dateAdd": "2022-06-23T19:14:04+00:00"
                }    
    ```
</details>

<details>
<summary>CLI tools agent_control :red_circle: </summary>

1. Executed on Manager:
    ```
    /var/ossec/bin/agent_control -i AGENT_ID
    ```
    Output:

    ```
    Wazuh agent_control. Agent information:
       Agent ID:   002
       Agent Name: macos-1201
       IP address: any
       Status:     Active
    
       Operating system:    Darwin |macos-1201 |21.1.0 |Darwin Kernel Version 21.1.0: Wed Oct 13 17:33:23 PDT 2021; root:xnu-8019.41.5~1/RELEASE_X86_64 |x86_64
       Client version:      Wazuh v4.4.0
       Configuration hash:  ab73af41699f13fdd81903b5f23d8d00
       Shared file hash:    4a8724b20dee0124ff9656783c490c4e
       Last keep alive:     1656012344
    
       Syscheck last started at:  Thu Jun 23 19:14:42 2022
       Syscheck last ended at:    Thu Jun 23 19:14:53 2022
    
    ```
</details>

**Comment by jnasselle** [comment]
Hi @CamiRomero ,

As we have discussed, `agent_control` output needs to be improved and has nothing to do with changes proposed in the PR, so I'v created this issue to do this https://github.com/wazuh/wazuh/issues/14030

Regarding Wazuh Dashboard, you're right: there's no codename display in any part at all. IMHO, the codename on MacOS is more used than the version number and could be shown also as part of OS column.

In conclusion: both reds on agent_control CLI and Wazuh Dashboard are unrelated to the proposed change.

**Comment by jnasselle** [comment]
Hi @CamiRomero ,

Related PR was re-targeted to v4.3.6, so issue description, Milestone and label was updated

**Comment by CamiRomero** [comment]
As @jnasselle mentions in [comment](https://github.com/wazuh/wazuh-qa/issues/3000#issuecomment-1166705307) both reds on agent_control CLI and Wazuh Dashboard are unrelated to the proposed change.

2 new improvement issues were defined to deal with both cases:

- [Improve Operating System information retrieved by agent_control tool](https://github.com/wazuh/wazuh/issues/14030)
